OpenRA.Mods.Common/Traits/World/ResourceLayer.cs

```diff
@@ -66,14 +66,18 @@ public void WorldLoaded(World w, WorldRenderer wr)
 			content = new CellLayer<CellContents>(w.Map);
 			render = new CellLayer<CellContents>(w.Map);
 			dirty = new List<CPos>();
+			Update();
+		}
 
-			var resources = w.WorldActor.TraitsImplementing<ResourceType>()
+		public void Update()
```

Comment:
ResourceLayer should register a `CellEntryChanged` callback against MapResources instead of coupling this to the editor.